rskj-core/src/main/java/org/ethereum/db/BlockStore.java

```diff
@@ -41,11 +41,18 @@
 
     Block getChainBlockByNumber(long blockNumber);
 
-    @Nonnull
     List<Block> getChainBlocksByNumber(long blockNumber);
 
     void removeBlock(Block block);
 
+    /**
+     * Retrieves a block header by its hash.
+     *
+     * @param hash The hash of the block to look up, cannot be null.
+     * @return The block header if found, null if not.
+     */
+    BlockHeader getBlockHeaderByHash(byte[] hash);
```

Comment:
New methods such as this one should use the correct API: hashes are represented by `Keccak256` instances.

It's OK to leave old method unchanged for now if that is not related with your PR.